### PR TITLE
Decimal priorities-Sequential Subject selection

### PR DESCRIPTION
**Pull request by yuenmichelle1**
Ensure we correctly sort decimal numbers with a Mantissa for prioritized workflows. Before this PR the `List.keysort` function didn't correctly sort the decimal type values due to the underlying module struct values, see this issue for more details https://github.com/ericmj/decimal/issues/45

As such this PR adds the following to ensure we correctly order all decimal values
- updates the Decimal package that Ecto uses 
- switches to a custom `Enum.sort_by` function from `List.keysort` using the `Decimal.cmp` package functions
- updates the sql query to add a `case when` to ensure the resulting tuples have a valid decimal value (not nil) for sorting on

### CODE REVIEW

**Comment by camallen** [comment]
@yuenmichelle1 mind having a look over this - as it's your PR I don't think you can review it but I think it's worth getting some more eyeballs on it.

File: web/models/workflow.ex [hunk 1]
```diff
@@ -40,4 +40,14 @@ defmodule Designator.Workflow do
     |> cast(params, [:id, :configuration, :grouped, :prioritized])
   end
 
+  def sort_decimal_values(result_tuples) do
+    # sort_by/3 takes the following params
+    # result set param for sorting
+    # a mapper function to unpack the tuple index [1] decimal values
+    # and a comparison function to sort on ensuring ascending order (elem 1 < elem 2)
+    # https://hexdocs.pm/decimal/1.9.0/readme.html#comparisons
+    Enum.sort_by(result_tuples, &elem(&1, 1), &Decimal.cmp(&1, &2) != :gt)
```
Comment (yuenmichelle1) [inline web/models/workflow.ex:49]: Ah this was something I was looking into. It looked like Decimal comparison was fixed in the most recent version 2.0.0, but I ran into dependency conflict issues. nice.

File: web/models/workflow.ex [hunk 2]
```diff
@@ -29,9 +29,9 @@ defmodule Designator.Workflow do
     query = from sms in "set_member_subjects",
       left_join: swc in "subject_workflow_counts", on: (sms.subject_id == swc.subject_id and swc.workflow_id == ^workflow_id),
       where: sms.subject_set_id == ^subject_set_id and is_nil(swc.retired_at),
-      select: {sms.subject_id, sms.priority}
+      select: { sms.subject_id, fragment("CASE WHEN priority IS NULL THEN 0 ELSE priority END") }
```
Comment (yuenmichelle1) [inline web/models/workflow.ex:32]: I'm assuming the `fragment` part is to take care of the case when priority is `nil`?  Does the sort/Decimal compare have issues when comparing against `nil`?

**Review by yuenmichelle1 — COMMENTED** [review]
@camallen  All this looks pretty good to me. Thanks for looking into this :) I had some general questions, just things I was curious about. 

Also curious, how often is it that we get `priority` being decimals? (ISTROX was the first time I've encountered, but I was under the assumption `priority` would always be integers?)

**Comment by yuenmichelle1** [comment]
I tested this with ISTROX locally (by hooking up readonly db to local running designator) and the output ids are matching what cellect is also outputting. I also tested a project where set_member_subjects has integer `priority` and it works as expected as well 👍 🎉  🥂

**Pull request merged**